packages/datadog-plugin-jest/src/index.js

```diff
@@ -0,0 +1,197 @@
+const { exec } = require('child_process')
+const { promisify } = require('util')
+
+const promiseExec = promisify(exec)
+
+const GIT_COMMIT_SHA = 'git.commit_sha'
+const GIT_BRANCH = 'git.branch'
+const GIT_REPOSITORY_URL = 'git.repository_url'
+const BUILD_SOURCE_ROOT = 'build.source_root'
+const TEST_FRAMEWORK = 'test.framework'
+const TEST_TYPE = 'test.type'
+const TEST_NAME = 'test.name'
+const TEST_SUITE = 'test.suite'
+const TEST_STATUS = 'test.status'
+const CI_PIPELINE_URL = 'ci.pipeline.url'
+const CI_PIPELINE_ID = 'ci.pipeline.id'
+const CI_PIPELINE_NUMBER = 'ci.pipeline.number'
+const CI_WORKSPACE_PATH = 'ci.workspace_path'
+const CI_PROVIDER_NAME = 'ci.provider.name'
+
+function getCIMetadata () {
+  const { env } = process
+  if (env.GITHUB_ACTIONS) {
+    const { GITHUB_REF, GITHUB_SHA, GITHUB_REPOSITORY, GITHUB_RUN_ID, GITHUB_RUN_NUMBER, GITHUB_WORKSPACE } = env
+
+    const pipelineURL = `https://github.com/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}`
+
+    return {
+      [CI_PIPELINE_URL]: pipelineURL,
+      [CI_PIPELINE_ID]: GITHUB_RUN_ID,
+      [CI_PROVIDER_NAME]: 'github',
+      [CI_PIPELINE_NUMBER]: GITHUB_RUN_NUMBER,
+      [CI_WORKSPACE_PATH]: GITHUB_WORKSPACE,
+      [GIT_BRANCH]: GITHUB_REF,
+      [GIT_COMMIT_SHA]: GITHUB_SHA
+    }
+  }
+  return {}
+}
+
+async function getGitInformation () {
+  const sanitizeStdout = (stdout) => stdout.replace(/(\r\n|\n|\r)/gm, '')
+  try {
+    const [{ stdout: readRepository }, { stdout: readBranch }, { stdout: readCommit }] = await Promise.all([
+      promiseExec('git ls-remote --get-url'),
+      promiseExec('git branch --show-current'),
+      promiseExec('git rev-parse HEAD')
+    ])
+    return {
+      repository: sanitizeStdout(readRepository),
+      branch: sanitizeStdout(readBranch),
+      commit: sanitizeStdout(readCommit)
+    }
+  } catch (e) {
+    return {
+      repository: '',
+      branch: '',
+      commit: ''
+    }
+  }
+}
+
+function finishStartedSpans () {
+  global.tracer
+    .scope()
+    .active()
+    ._spanContext._trace.started.forEach((span) => {
+      span.finish()
+    })
+}
+
+function setTestStatus (status) {
+  global.tracer.scope().active().setTag(TEST_STATUS, status)
+}
+
+function addTestMetadata ({ testName, testSuite }) {
```

Comment:
This function can be omitted, because in every case, you know these values at the time of the span creation, so you can just add them to the `tags` property of the options object.

Reply:
I thought of that but I think I couldn't do it 🤔 . I will try again though 👍 , I probably messed something up